From b1b78b36bba9f1218ecb7e3f345cc6e1a604c2ed Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 00:58:35 -0400
Subject: [acpi] dock: drop redundant spin lock in dock station object

Message-id: <20140710005834.22146.81762.stgit@gir.stowe>
Patchwork-id: 85542
O-Subject: [RHEL7.1 PATCH 003/105] ACPI / dock: drop redundant spin lock in dock station object
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: ed633e709fa633c6bc24f4d6ecb55ad0c14fd335

commit ed633e709fa633c6bc24f4d6ecb55ad0c14fd335
Author: Jiang Liu <jiang.liu@huawei.com>
Date:   Sat Jun 29 00:24:35 2013 +0800

    ACPI / dock: drop redundant spin lock in dock station object

    All dock station objects are created during initialization and
    don't change at runtime, so drop the redundant spin lock from
    struct dock_station.

    [rjw: Changelog]
    Signed-off-by: Jiang Liu <jiang.liu@huawei.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/dock.c b/drivers/acpi/dock.c
index c36de86..750f958 100644
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -63,7 +63,6 @@ struct dock_station {
  acpi_handle handle;
  unsigned long last_dock_time;
  u32 flags;
- spinlock_t dd_lock;
  struct mutex hp_lock;
  struct list_head dependent_devices;
 
@@ -112,10 +111,7 @@ add_dock_dependent_device(struct dock_station *ds, acpi_handle handle)
 
  dd->handle = handle;
  INIT_LIST_HEAD(&dd->list);
-
- spin_lock(&ds->dd_lock);
  list_add_tail(&dd->list, &ds->dependent_devices);
- spin_unlock(&ds->dd_lock);
 
  return 0;
 }
@@ -220,14 +216,10 @@ find_dock_dependent_device(struct dock_station *ds, acpi_handle handle)
 {
  struct dock_dependent_device *dd;
 
- spin_lock(&ds->dd_lock);
- list_for_each_entry(dd, &ds->dependent_devices, list) {
-  if (handle == dd->handle) {
-   spin_unlock(&ds->dd_lock);
+ list_for_each_entry(dd, &ds->dependent_devices, list)
+  if (handle == dd->handle)
    return dd;
-  }
- }
- spin_unlock(&ds->dd_lock);
+
  return NULL;
 }
 
@@ -1005,7 +997,6 @@ static int __init dock_add(acpi_handle handle)
  dock_station->last_dock_time = jiffies - HZ;
 
  mutex_init(&dock_station->hp_lock);
- spin_lock_init(&dock_station->dd_lock);
  INIT_LIST_HEAD(&dock_station->sibling);
  INIT_LIST_HEAD(&dock_station->dependent_devices);
 
-- 
1.7.1